### ISSUE TITLE
Can receive but can't send email

### ISSUE THREAD

**Issue by Shagon94** [mailcow/mailcow-dockerized#2115]
**Notice (you can delete this paragraph)**
Please understand that we use GitHub as **bug tracker and for feature requests only**.
For general (community) support and other discussion, you are welcome to visit us @ Freenode, #mailcow
Answering may take from a few seconds to hours, please be patient. :-)
Immediate commercial support is available via info@servercow.de

**Describe the bug**
I can receive emails but cannot send them

**How to reproduce Reproduce**
Steps to reproduce the behavior please explain how to reproduce the bug:
send an email - nothing
receiving works as expected

**Expected behavior**
When sending an email the recipient should receive the email

**Screenshots**
self explanatory 

**Logs**
Please take a look at the [documentation](https://mailcow.github.io/mailcow-dockerized-docs/debug-logs/).
Checked the documentation however the logs are huge, is there any specific service that I should be looking for?

**System**
Operating System: Ubuntu 16.04.5 LTS
Kernel: Linux 4.15.0-42-generic
 - Browser: [e.g. chrome, safari] any
 - Other clients involved: any

**Additional context**
Telnet works:

```
$ telnet 51.15.93.189 587
Trying 51.15.93.189...
Connected to 51.15.93.189.
Escape character is '^]'.
220 mail.whitepulsar.space ESMTP Postcow
^]

$ telnet 51.15.93.189 465
Trying 51.15.93.189...
Connected to 51.15.93.189.
Escape character is '^]'.
^]
```

mxtoolbox shows everything should be in order - https://mxtoolbox.com/SuperTool.aspx?action=mx%3awhitepulsar.space&run=toolpage

the machine has enough ram and cpu power albeit that isn't the issue

the IP address isn't blacklisted on https://www.dnsbl.info/ 

I've also commented on IRC however due to timezones this question might not get answered, I am happy to get a license or donate or anything if the issue is resolved since I will be needing to do this setup once more on 2-3 more servers for my friends, would really like to have this resolved. Thank you.

**Comment by andryyy** [comment]
Hi, you will need to post logs. Postfix logs would be important.

**Comment by Shagon94** [comment]
Hello @andryyy,

Thank you for the promptly response. I will get the logs and update as soon as I can, I've left my home to pickup some things, merry Christmas by the way to you and the team.

I'll check the postfix logs and try to make it work, hopefully they will give further insight into this.

Kind Regards.

**Comment by Shagon94** [comment]
Hello, after some deep investigation I've concluded the following:

checking the certificates:

* mail.whitepulsar.space:25:

```bash
# echo | openssl s_client -starttls smtp -crlf -connect mail.whitepulsar.space:25 2>/dev/null | openssl x509 -subject -dates -issuer -noout
subject=CN = mail.whitepulsar.space
notBefore=Dec 22 08:03:26 2018 GMT
notAfter=Mar 22 08:03:26 2019 GMT
issuer=C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
```

* mail.whitepulsar.space:465
```bash
# echo | openssl s_client -showcerts -connect mail.whitepulsar.space:465 2>/dev/null | openssl x509 -subject -dates -issuer -noout
subject=CN = mail.whitepulsar.space
notBefore=Dec 22 08:03:26 2018 GMT
notAfter=Mar 22 08:03:26 2019 GMT
issuer=C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
```

* mail.whitepulsar.space:587:

```bash
# echo | openssl s_client -starttls smtp -crlf -connect mail.whitepulsar.space:587 2>/dev/null | openssl x509 -subject -dates -issuer -noout
subject=CN = mail.whitepulsar.space
notBefore=Dec 22 08:03:26 2018 GMT
notAfter=Mar 22 08:03:26 2019 GMT
issuer=C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
```

Certificates are valid.

postfix-mailcow logs:

```bash
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:37:49 mail postfix/smtpd[688]: connect from mailcowdockerized_sogo-mailcow_1_2570e785efd3.mailcowdockerized_mailcow-network[172.22.1.248]
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:37:49 mail postfix/smtpd[688]: A012F84C14: client=mailcowdockerized_sogo-mailcow_1_2570e785efd3.mailcowdockerized_mailcow-network[172.22.1.248], sasl_method=PLAIN, sasl_username=admin@whitepulsar.space
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:37:49 mail postfix/cleanup[693]: A012F84C14: message-id=<47-5c1e2200-7-43059900@90316927>
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:37:57 mail postfix/qmgr[347]: A012F84C14: from=<admin@whitepulsar.space>, size=1682, nrcpt=1 (queue active)
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:37:57 mail postfix/smtpd[688]: disconnect from mailcowdockerized_sogo-mailcow_1_2570e785efd3.mailcowdockerized_mailcow-network[172.22.1.248] ehlo=1 auth=1 mail=1 rcpt=1 data=1 quit=1 commands=6
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:38:28 mail postfix/smtp[694]: connect to gmail-smtp-in.l.google.com[108.177.126.27]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:38:29 mail postfix/smtp[694]: connect to gmail-smtp-in.l.google.com[2a00:1450:4013:c01::1a]:25: Network is unreachable
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:38:59 mail postfix/smtp[694]: connect to alt1.gmail-smtp-in.l.google.com[74.125.24.27]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:39:00 mail postfix/smtp[694]: connect to alt1.gmail-smtp-in.l.google.com[2404:6800:4003:c03::1b]:25: Network is unreachable
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:39:30 mail postfix/smtp[694]: connect to alt2.gmail-smtp-in.l.google.com[108.177.125.26]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 12:39:30 mail postfix/smtp[694]: A012F84C14: to=<myEmailAddress@gmail.com>, relay=none, delay=101, delays=8.2/0.05/93/0, dsn=4.4.1, status=deferred (connect to alt2.gmail-smtp-in.l.google.com[108.177.125.26]:25: Connection timed out)
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:32:36 mail postfix/smtp[824]: connect to mx1.mail.icloud.com[17.172.34.10]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:32:36 mail postfix/smtp[816]: connect to mx4.mail.icloud.com[17.133.229.9]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:32:37 mail postfix/smtp[823]: connect to alt1.gmail-smtp-in.l.google.com[74.125.24.26]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:32:38 mail postfix/smtp[823]: connect to alt1.gmail-smtp-in.l.google.com[2404:6800:4003:c03::1b]:25: Network is unreachable
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:33:06 mail postfix/smtp[824]: connect to mx5.mail.icloud.com[17.57.8.138]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:33:07 mail postfix/smtp[816]: connect to mx2.mail.icloud.com[17.142.163.14]:25: Connection timed out
postfix-mailcow_1_b8ffccd294c7 | Dec 22 13:34:06 mail postfix/smtp[824]: 2A78884D4F: to=<myEmailAddress@icloud.com>, relay=none, delay=12631, delays=12481/0.14/150/0, dsn=4.4.1, status=deferred (connect to mx4.mail.icloud.com[17.178.97.77]:25: Connection timed out)
```

IP Tables:

```bash
# iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
MAILCOW    all  --  anywhere             anywhere
ACCEPT     udp  --  anywhere             anywhere             udp dpt:openvpn

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
MAILCOW    all  --  anywhere             anywhere
DOCKER-USER  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-1  all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  10.8.0.0/24          anywhere
ACCEPT     all  --  anywhere             anywhere             state RELATED,ESTABLISHED

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain DOCKER (3 references)
target     prot opt source               destination
ACCEPT     udp  --  anywhere             172.17.0.2           udp dpt:openvpn
ACCEPT     tcp  --  anywhere             172.22.1.2           tcp dpt:12345
ACCEPT     tcp  --  anywhere             172.22.1.2           tcp dpt:sieve
ACCEPT     tcp  --  anywhere             172.22.1.2           tcp dpt:pop3s
ACCEPT     tcp  --  anywhere             172.22.1.2           tcp dpt:imaps
ACCEPT     tcp  --  anywhere             172.22.1.2           tcp dpt:imap2
ACCEPT     tcp  --  anywhere             172.22.1.3           tcp dpt:mysql
ACCEPT     tcp  --  anywhere             172.22.1.2           tcp dpt:pop3
ACCEPT     tcp  --  anywhere             172.22.1.6           tcp dpt:submission
ACCEPT     tcp  --  anywhere             172.22.1.6           tcp dpt:urd
ACCEPT     tcp  --  anywhere             172.22.1.6           tcp dpt:smtp
ACCEPT     tcp  --  anywhere             172.22.1.10          tcp dpt:11443
ACCEPT     tcp  --  anywhere             172.22.1.10          tcp dpt:11080

Chain DOCKER-ISOLATION-STAGE-1 (1 references)
target     prot opt source               destination
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-ISOLATION-STAGE-2 (3 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-USER (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere

Chain MAILCOW (2 references)
target     prot opt source               destination
```

Checking outbound traffic:

```bash
# telnet gmail-smtp-in.l.google.com 25
Trying 108.177.126.27...
Trying 2a00:1450:4013:c01::1a...
telnet: Unable to connect to remote host: Network is unreachable

# telnet alt1.gmail-smtp-in.l.google.com 25
Trying 74.125.24.27...
Trying 2404:6800:4003:c03::1a...
telnet: Unable to connect to remote host: Network is unreachable

# telnet mx5.mail.icloud.com 25
Trying 17.172.34.68...
Trying 17.172.34.69...
Trying 17.133.229.13...
Trying 17.133.229.11...
Trying 17.178.97.78...
Trying 17.178.97.74...
Trying 17.57.8.143...
Trying 17.57.8.138...
telnet: Unable to connect to remote host: Network is unreachable
```

I'm guessing outbound traffic is blocked on that server since I can connect to it from outside:

```bash
~ # echo exit | telnet mail.whitepulsar.space 25
Trying 51.15.93.189...
Connected to mail.whitepulsar.space.
Escape character is '^]'.
Connection closed by foreign host.
```

And there is an entry towards google's nameservers on the machine:

```bash
# bat /etc/resolv.conf -p
domain cloud.online.net
search cloud.online.net
nameserver 8.8.8.8
nameserver 10.6.30.8
nameserver 10.6.30.9
```

However I did not use a public resolver in the `data/conf/unbound/unbound.conf`:

```bash
# grep forward data/conf/unbound/unbound.conf

```

because of:

>Please do not use a public resolver like we did in the example above. Many - if not all - blacklist lookups will fail with public resolvers.
taken from - https://mailcow.github.io/mailcow-dockerized-docs/u_e-unbound-fwd/

When checking this theory on a digital ocean instance:

```bash
~ # echo exit | telnet alt1.gmail-smtp-in.l.google.com 25
Trying 74.125.24.26...
Connected to alt1.gmail-smtp-in.l.google.com.
Escape character is '^]'.
Connection closed by foreign host.

~ # echo exit | telnet gmail-smtp-in.l.google.com 25
Trying 108.177.119.27...
Connected to gmail-smtp-in.l.google.com.
Escape character is '^]'.
Connection closed by foreign host.

~ # echo exit | telnet mx5.mail.icloud.com 25
Trying 17.172.34.69...
Connected to mx5.mail.icloud.com.
Escape character is '^]'.
Connection closed by foreign host.
```

We get connected to the server in question (note - mailcow isn't installed on this server).

With this data in mind, can you confirm if the issue is regarding the server instance and not the software itself?

Kind Regards.

**Comment by andryyy** [comment]
Outbound traffic is blocked, yes.

**Comment by Shagon94** [comment]
Hello @andryyy 

Thank you for confirming. Will reach out to my server provider to get a resolution on this.

Kind Regards.

**Comment by Shagon94** [comment]
Hello team,

Thank you for helping me out on this.

This has been resolved, the kind people at ScaleWay have removed the restriction.

I'll close this one, thanks for the quick reply.

Kind Regards.

**Comment by NoobKeksTV** [comment]
Ihave the same issue. 
My Server in general works fine. Not Traffic is blocked. i believe i messed up my iptables setup. 
I can receive any email but cant send any. It did work tho before i messed with iptables some while ago.

My Iptables rules:
https://screens.egopvp.com/files/2020/09/25/putty_bJVa9ePaB8.png

**Comment by MAGICCC** [comment]
Can you please stop hijacking other issues?
Since GitHub is not a support forum either ask in our [telegram channel](https://telegram.me/mailcow) or in [our forum](https://community.mailcow.email/)
